lib/rubocop/cop/lint/detect_invalid_pack_directives.rb

```diff
@@ -0,0 +1,196 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      class DetectInvalidPackDirectives < RuboCop::Cop::Base
+        MODIFIABLE_DIRECTIVES = %w[s S i I l L q Q j J]
+        ACCEPTABLE_DIRECTIVES = %w[U m A B H a A Z b B h H c C s S i I l L q Q j J n N v V f F e E d D g G x X @ % U u m M P p w]
+
+        # @param [RuboCop::AST::SendNode] node Node for the ruby `send` method
+        # @return [[RuboCop::AST::Node], raise] offense when an invalid directive is found, or raise if unexpected error found
+        def on_send(node)
+          _callee, method_name = *node
+
+          return unless %i[pack unpack pack1 unpack1].include?(method_name)
+
+          args = node.arguments
+          return if args.empty?
+
+          args.each do |arg|
+            next unless string_arg?(arg)
+
+            # if multiline arguments are passed
+            if arg.type == :dstr
+              idx = []
+
+              pack_directive = arg.children.map do |child|
+                if begin_arg?(child)
+                  next
+                else
+                  idx << child.children.first.length
+                  child.children.join
+                end
+              end.join
+
+            # elsif single line arguments are passed
+            elsif arg.type == :str
+              pack_directive = arg.children.first
+            end
+
+            error = validate_directive(pack_directive)
+            if error.nil?
+              next
+            else
+              offense_range = get_error_range(arg, error[:index], idx)
+              return if offense_range.nil?
+
+              add_offense(offense_range, message: error[:message])
+            end
+          rescue StandardError => e
+            raise e
+          end
+        end
+
+        private
+
+        # Check if the pack directives are valid. See link for pack docs https://apidock.com/ruby/Array/pack
+        #
```

Comment:
Can we add a permalink to the code that you were implementing here

Reply:
As in this [PR](https://github.com/rapid7/metasploit-framework/pull/18187)?

Reply:
https://github.com/ruby/ruby/blob/6391132c03ac08da0483adb986ff9a54e41f9e14/pack.c#L196